server/index.js

```diff
@@ -2,10 +2,81 @@ require('dotenv/config');
 const express = require('express');
 const staticMiddleware = require('./static-middleware');
 const errorMiddleware = require('./error-middleware');
+const pg = require('pg');
+const argon2 = require('argon2');
+const jwt = require('jsonwebtoken');
+const ClientError = require('./client-error');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
 
 const app = express();
 
 app.use(staticMiddleware);
+app.use(express.json());
+
+app.post('/api/auth/register', (req, res, next) => {
+  const { email, password } = req.body;
+  if (!email || !password) {
+    throw new ClientError(400, 'email and password are required fields');
+  }
```

Comment:
Since you have password requirements on the client side you need to validate the password meets the requirements here as well. It is more important that the server validates the password since people can send requests to your server without using your UI. Data validation must be done server side. Client side validation is just good UX but won't actually protect your data.